typescript.md

```diff
@@ -30,15 +30,29 @@ defines rules for TypeScript specific features.
 `tsc` has some [options](http://www.typescriptlang.org/docs/handbook/compiler-options.html)
 to make type checks stricter:
 
-* `noImplicitAny`;
-* `noImplicitReturns`;
-* `noUnusedLocals`;
-* `noUnusedParameters`;
-* `noImplicitThis`;
-* `strictNullChecks`.
-
-It's recommended to turn them on as early as possible in development (and migration
-from JS).
+```json
+{
+    "compilerOptions": {
+        "strict": true,
+        "noImplicitReturns": true,
```

Comment:
also, I want to remove this option, because the following approach 
```ts
function f(flag: boolean):  number | undefined {
    if (flag) {
        return 1;
    }
}
```

is very common is JavaScript and **correctly checked** with TypeScript. So, I don't understand why I need explicitly specify `return` statement:

```diff
function f(flag: boolean):  number | undefined {
    if (flag) {
        return 1;
    } 
+   return undefined;
}
```